vizro-core/src/vizro/models/_components/card.py

```diff
@@ -36,8 +35,8 @@ class Card(VizroBaseModel):
     def build(self):
         text = dcc.Markdown(self.text, className="card_text", dangerously_allow_html=False, id=self.id)
         button = html.Div(
-            dbc.Button(
-                href=get_relative_path(self.href) if self.href.startswith("/") else self.href, className="card_button"
+            dcc.Link(
+                href=get_relative_path(self.href) if self.href.startswith("/") else self.href, className="card-link"
```

Comment:
It made more sense semantically here to convert it to a `dcc.Link`

Reply:
This is massively better now but still feels hacky to me. Why can't we do it in what I would consider to be the "correct" way like this:
```
dcc.Link(text, href=...) if self.href is not None else text
```

Either way, we should change the variable names and class name so it's no longer `button`.

Reply:
As for the navigation, our current solution feels quite abusive to me and so if we can fix it I think we should (doesn't have to be this PR though).

1. For now, how about setting `color="link"` on the `dbc.Button`? See https://dash-bootstrap-components.opensource.faculty.ai/docs/components/button/ for examples. This feels like it would be the right solution to me. Considering that the docs on `dbc.Accordion` include buttons inside the accordion, I don't think that's a naughty thing to do so long as it's clear that the button is operating as a link
2. If that doesn't work well, what about instead just using `dcc.Link`, dropping the `key` argument (unless I misunderstood something, it doesn't feel particularly important here) and doing the `active` styling with CSS classname instead 
3. Way outside scope for this PR, but possibly in future we should use [`NavLink`](https://dash-bootstrap-components.opensource.faculty.ai/docs/components/nav/) inside a proper `Nav`. This feels most correct and actually supports a more advanced `active` than `dbc.Button`

Reply:
I'll respond to your first request first, as the second one I need to try out first.

Yes, that's a great idea actually! That also works perfectly fine, I've implemented the change [here](https://github.com/mckinsey/vizro/pull/311/commits/ca213993b2da2a562c71832519353c1638eefb4c) and also got rid of one of the redundant htmls! Do you think it's worth implementing the CSS name changing from underscore to dashes already? Or better to do in one separate PR? Or not doing at all if we at some point switch to BEM / Bootstrap syntax?

Reply:
On your second question:
1. Should be do-able and I could implement this in the current PR already.
2. I slightly remember @nadijagraca actually trying that out when she did implement the navigations. She also tried out the active CSS property as far as I remember, but it didn't seem to work? @nadijagraca - can you confirm? or did you not try that out? 
3. @nadijagraca - any reason why we didn't use NavLink in the first place? I somehow also remember there was something 😄

Reply:
2. I did try out the `active` property, but it does not cover fully our use case. We need the styling to be permanent as long as the user is on that page. When using `active` property, the styling would go away as soon as you click anywhere on the page.  
3. I remember there were some problem with it in the old repo, but I cannot recall what was the issue. I didn't try implementing it in the new repo.

Reply:
@antonymilne @nadijagraca - I actually just tried out the NavLink in the Accordion and that seems to work perfectly with the active CSS property only! This means we can actually replace the dbc.Buttons with dbc.NavLinks and could even remove the page-id argument from the builds because the highlighting of the active NavLink could now be achieved via CSS only. I would split this into a separate PR though that I can do next @antonymilne - so would leave the current version unchanged.

Reply:
> Yes, that's a great idea actually! That also works perfectly fine, I've implemented the change [here](https://github.com/mckinsey/vizro/pull/311/commits/ca213993b2da2a562c71832519353c1638eefb4c) and also got rid of one of the redundant htmls! Do you think it's worth implementing the CSS name changing from underscore to dashes already? Or better to do in one separate PR? Or not doing at all if we at some point switch to BEM / Bootstrap syntax?

Thank you, this looks much better 👍 If you think the underscore to dashes is a change worth doing regardless of whether/when we switch to BEM/Bootstrap then let's do it in a separate PR - I don't really mind if we make the change but have a weak preference in favour of doing so if it doesn't take us any further away from BEM/Bootstrap, which I don't think it does.

> @antonymilne @nadijagraca - I actually just tried out the NavLink in the Accordion and that seems to work perfectly with the active CSS property only! 🥳 This means we can actually replace the dbc.Buttons with dbc.NavLinks and could even remove the page-id argument from the builds because the highlighting of the active NavLink could now be achieved via CSS only. I would split this into a separate PR though that I can do next @antonymilne - so would leave the current version unchanged.

This is great news. Definitely let's do it in a separate PR since I'll have a couple more comments on it I think.